courier/courier.go

```diff
@@ -51,26 +46,35 @@ type (
 	}
 
 	courier struct {
-		smsClient           *smsClient
-		smtpClient          *smtpClient
-		httpClient          *httpClient
+		courierChannels     map[string]Channel
 		deps                Dependencies
 		failOnDispatchError bool
 		backoff             backoff.BackOff
 	}
 )
 
 func NewCourier(ctx context.Context, deps Dependencies) (Courier, error) {
-	smtp, err := newSMTP(ctx, deps)
+	emailChannel, err := NewEmailChannel(ctx, deps)
 	if err != nil {
 		return nil, err
 	}
+	return NewCourierWithCustomEmailChannel(ctx, deps, emailChannel)
+}
+
+func NewCourierWithCustomEmailChannel(ctx context.Context, deps Dependencies, channel *EmailChannel) (Courier, error) {
+	channels := make(map[string]Channel, len(deps.CourierConfig().CourierChannels(ctx)))
+	for _, c := range deps.CourierConfig().CourierChannels(ctx) {
+		// TODO: add support for more channel types (e.g. SMTP, ..?)
```

Comment:
TODO?

Reply:
Yea, @zepatrik suggested that we replace the current config structure with a "channel" based system, that allows you to specify which channel the verification should go through in the identity schema (the `"via"` key). And to still support SMTP we'll need to add a "channel" type config key, but for now only the phone channel is supported this way. I'd work on that, once SMS is fully done.